barretenberg/cpp/src/barretenberg/flavor/ultra_rollup_flavor.hpp

```diff
@@ -103,20 +97,18 @@ class UltraRollupFlavor : public bb::UltraFlavor {
             }
         }
 
-        VerificationKey(ProvingKey& proving_key)
-            : ipa_claim_public_input_key(proving_key.ipa_claim_public_input_key)
+        VerificationKey(ProverPolynomials& polynomials, const auto& metadata)
         {
-            this->circuit_size = proving_key.circuit_size;
+            this->circuit_size = metadata.circuit_size;
             this->log_circuit_size = numeric::get_msb(this->circuit_size);
-            this->num_public_inputs = proving_key.num_public_inputs;
-            this->pub_inputs_offset = proving_key.pub_inputs_offset;
-            this->pairing_inputs_public_input_key = proving_key.pairing_inputs_public_input_key;
-
-            if (!proving_key.commitment_key.initialized()) {
-                proving_key.commitment_key = CommitmentKey(proving_key.circuit_size);
-            }
-            for (auto [polynomial, commitment] : zip_view(proving_key.polynomials.get_precomputed(), this->get_all())) {
-                commitment = proving_key.commitment_key.commit(polynomial);
+            this->num_public_inputs = metadata.num_public_inputs;
+            this->pub_inputs_offset = metadata.pub_inputs_offset;
+            this->pairing_inputs_public_input_key = metadata.pairing_inputs_public_input_key;
+            this->ipa_claim_public_input_key = metadata.ipa_claim_public_input_key;
```

Comment:
This wasn't here before so I guess its not needed. These should all be going away soon anyway